agent/utils/cpuinfo.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package utils
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/aws/amazon-ecs-agent/agent/utils/ioutilwrapper"
+)
+
+type CPUInfo struct {
+	Processors []Processor `json:"processors"`
+}
+
+type Processor struct {
+	Flags []string `json:"flags"`
+}
+
+var infoRegExp = regexp.MustCompile("([^:]*?)\\s*:\\s*(.*)$")
```

Comment:
Can you add a sample line and some explanatory text

Reply:
I will get rid of regex to make it easy to read and maintain